adapters/repos/db/lsmkv/strategies_map.go

```diff
@@ -249,6 +274,31 @@ func (kv MapPair) Bytes() ([]byte, error) {
 	return out.Bytes(), nil
 }
 
+func (kv *MapPair) BytesInverted() ([]byte, error) {
+	// make sure the 2 byte length indicators will never overflow:
+	if len(kv.Key) >= math.MaxUint16 {
+		return nil, errors.Errorf("mapCollection key must be smaller than %d",
+			math.MaxUint16)
+	}
+
+	if len(kv.Value) >= math.MaxUint16 {
+		return nil, errors.Errorf("mapCollection value must be smaller than %d",
+			math.MaxUint16)
+	}
+
+	out := bytes.NewBuffer(nil)
+
+	if _, err := out.Write(kv.Key); err != nil {
```

Comment:
key length is not encoded but `FromBytesInverted` seems to be expecting a fixed size key of 8 bytes length. If that's the case then the key len validation in this method could validate exactly that

Reply:
You are right, keys and values are always 8 bytes long, I'll add more validation